### PR TITLE
DiceRoller: Add R8 release config with Proguard files

### PR DESCRIPTION
**Pull request by cartland**
https://developer.android.com/build/shrink-code#enable

### CODE REVIEW

**Comment by cartland** [comment]
I tried to reproduce https://github.com/android/kotlin-multiplatform-samples/issues/13. I think release builds work. The project had a few version updates that might have fixed any issues.

File: DiceRoller/androidApp/build.gradle.kts [hunk 1]
```diff
@@ -36,6 +36,28 @@ android {
...
+            isShrinkResources = true
+
+            proguardFiles(
+                // Includes the default ProGuard rules files that are packaged with
+                // the Android Gradle plugin. To learn more, go to the section about
+                // R8 configuration files.
+                getDefaultProguardFile("proguard-android-optimize.txt"),
+
+                // Includes a local, custom Proguard rules file
+                "proguard-rules.pro"
```
Comment (kkuan2011) [inline DiceRoller/androidApp/build.gradle.kts:57]: do we have this file?

File: DiceRoller/androidApp/build.gradle.kts [hunk 2]
```diff
@@ -36,6 +36,28 @@ android {
...
+
+            // Enables resource shrinking, which is performed by the
+            // Android Gradle plugin.
+            isShrinkResources = true
+
+            proguardFiles(
+                // Includes the default ProGuard rules files that are packaged with
+                // the Android Gradle plugin. To learn more, go to the section about
+                // R8 configuration files.
+                getDefaultProguardFile("proguard-android-optimize.txt"),
```
Comment (kkuan2011) [inline DiceRoller/androidApp/build.gradle.kts:54]: do we have this file in the project?

Comment (keyboardsurfer) [reply DiceRoller/androidApp/build.gradle.kts:54]: This is an automatically generated file that lives in the tools section and not the project itself.

File: DiceRoller/androidApp/build.gradle.kts (hunk 1 above)
Comment (keyboardsurfer) [reply DiceRoller/androidApp/build.gradle.kts:57]: I created one for DiceRoller. If it's not needed here we can remove this line.

Comment (kkuan2011) [reply DiceRoller/androidApp/build.gradle.kts:57]: do you mean you created one for Fruitties, Ben?

Comment (keyboardsurfer) [reply DiceRoller/androidApp/build.gradle.kts:57]: Yes, that's what I meant.

**Review by kkuan2011 — COMMENTED** [review]
LGTM

**Pull request merged**